batchproducer/batchproducer.go

```diff
@@ -363,11 +364,18 @@ func (b *batchProducer) sendBatch(batchSize int) int {
 
 	b.consecutiveErrors = 0
 	b.currentDelay = 0
-	succeeded := len(records) - res.FailedRecordCount
+	// note *int64 to int conversion - in practice we never expect 2 billion failed records
+	// in a single call since API only supports 500 records per call
+	var succeeded int
+	if res.FailedRecordCount != nil {
```

Comment:
this does the same action, then immediately below `res.FailedRecordCount` is nil checked again.

could do:
```
succeeded = len(records)
if not nil {
  succeeded = succeeded - (casting)
  log message from below
} else {
  log message from below
}